app/containers/HomePage/index.js

```diff
@@ -35,14 +33,25 @@ import saga from './saga';
 
 const key = 'home';
 
-export function HomePage({
-  username,
-  loading,
-  error,
-  repos,
-  onSubmitForm,
-  onChangeUsername,
-}) {
+const stateSelector = createStructuredSelector({
+  repos: makeSelectRepos(),
+  username: makeSelectUsername(),
+  loading: makeSelectLoading(),
+  error: makeSelectError(),
+});
+
+export default function HomePage() {
+  const { repos, username, loading, error } = useSelector(stateSelector);
+
+  const dispatch = useDispatch();
+
+  const onChangeUsername = evt => dispatch(changeUsername(evt.target.value));
+  const onSubmitForm = evt => {
```

Comment:
I think `onChangeUsername` and `onSubmitForm` can be wrapped/memoized with [`React.useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback) to avoid unnecessary re-renders.

Reply:
Might be pre-mature optimization to do this: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Thanks for the interesting read! You are probably correct about the performance difference here. I will try profiling this branch when I can. 

I still think it may be worth including for referential purposes (ex: some devs I work with only use the hooks that already exist in code -- easy copy+paste, I guess -- and never bother to read the Hooks API docs). In this context, I would suggest that boilerplate is synonymous with premature optimization.

Also, it appears I was wrong about `onSubmitForm` (didn't see that it wasn't being used as a prop). Definitely not supposed to wrap that one.